### DISCUSSION TITLE
Help with using Elasticsearch 8 with Quarkus

### DISCUSSION THREAD

**Discussion by nek4life** [quarkusio/quarkus#25843] [Community]
Looking for some guidance on how to use ES 8 with Quarkus. I'm starting a new app and looks like the current high level client is deprecated and the new open source client is for ES 8. I'd rather use the new client instead of building on top of the deprecated client. Any thoughts on the best way to accomplish this?

**Comment by Arsou33** [comment]
We are in a similar situation where we are currently evaluating the possibility of migrating a Quarkus based application from Elastic 7.10 to 8.
An alert has been posted in the ElasticSearch guide: 

- https://quarkus.io/guides/elasticsearch#using-the-high-level-rest-client

Still, it seems to me that the client is on an Apache 2 license: 

- https://github.com/elastic/elasticsearch-java/.

Is a Quarkus extension in progress or planned ?

#Edit: Even if licence file Elasticsearch client API is an Apache 2 file, every headers have the Elastic licence v2.

**Comment by ivangfr** [comment]
The workaround I've found to use ES 8

Added the following dependecy
```
<dependency>
    <groupId>co.elastic.clients</groupId>
    <artifactId>elasticsearch-java</artifactId>
    <version>8.5.3</version>
</dependency>
```

Set the application.properties
```
elasticsearch.hosts=http://${ELASTICSEARCH_HOST:localhost}:${ELASTICSEARCH_REST_PORT:9200}
```

Configure the ElasticsearchConfig
```
@ApplicationScoped
public class ElasticsearchConfig {

    @Produces
    @Singleton
    public ElasticsearchClient ElasticsearchClient(@ConfigProperty(name = "elasticsearch.hosts") String hosts) {
        RestClient restClient = RestClient.builder(HttpHost.create(hosts)).build();
        return new ElasticsearchClient(new RestClientTransport(restClient, new JacksonJsonpMapper()));
    }
}
```

Use the ElasticsearchClient in my service
```
@ApplicationScoped
public class MovieServiceImpl implements MovieService {

    @Inject
    ElasticsearchClient client;

    ...
}
```

**Comment by yrodiere** [comment]
The new Elasticsearch client has been merged and will be available in Quarkus 3: https://github.com/quarkusio/quarkus/pull/22622 . Documentation seems missing at the moment, though.

FWIW, there's also a Quarkiverse extension for OpenSearch clients: https://github.com/quarkiverse/quarkus-opensearch
